lib/modifiers.js

```diff
@@ -42,27 +42,12 @@ function groupRedact (o, path, censor, isCensorFct, censorFctTakesPath) {
 }
 
 function nestedRestore (arr) {
-  const length = arr.length
-  for (var i = 0; i < length; i++) {
-    const { key, target, value, level } = arr[i]
-    if (level === 0 || level === 1) {
-      if (has(target, key)) {
-        target[key] = value
-      }
-      /* istanbul ignore else */
-      if (typeof target === 'object') {
-        const targetKeys = Object.keys(target)
-        for (var j = 0; j < targetKeys.length; j++) {
-          const tKey = targetKeys[j]
-          const subTarget = target[tKey]
-          if (has(subTarget, key)) {
-            subTarget[key] = value
-          }
-        }
-      }
-    } else {
-      restoreNthLevel(key, target, value, level)
+  for (const { parent, path, value } of arr) {
```

Comment:
you may want to consider comparing the performance of this approach to iteration with a plain for loop. I suspect that the motivation behind the for loop in the original code was indeed related to performance.